### PR TITLE
APEX-37 Added APPLICATION_ATTEMPT_ID attribute and change operator and container history file so that different application attempts write to separate HDFS files

### PR DESCRIPTION
**Pull request by davidyan74**
This is in conjunction with another pull request in DTX:

https://github.com/DataTorrent/DTX/pull/237

### CODE REVIEW

**Comment by davidyan74** [comment]
Need to fix unit tests

**Comment by davidyan74** [comment]
test this please

File: engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java [hunk 1]
```diff
@@ -175,7 +177,7 @@
   private final Cache<Long, Object> commandResponse = CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.MINUTES).build();
   private long lastLatencyWarningTime;
   private transient ExecutorService poolExecutor;
-  private FileContext fileContext;
```
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java]: There is no need to replace FileContext with FileSystem.

**Comment by tweise** [comment]
Would like to understand why we need to write a file to HDFS for every physical operator, and that as part of the heartbeat RPC handling. This operation is expensive and there needs to be a good reason.

**Comment by davidyan74** [comment]
Thomas, all the items we discussed have been implemented.  Please merge this and DataTorrent/DTX#237 when you get a chance.  Thanks!

File: engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java [hunk 2]
```diff
@@ -1395,34 +1397,35 @@ public ContainerHeartbeatResponse processHeartbeat(ContainerHeartbeat heartbeat)
...
-          if (operatorFile == null) {
-            operatorFiles.putIfAbsent(ptOp.getId(), new FSJsonLineFile(new Path(this.vars.appPath + "/operators/" + ptOp.getId()), FsPermission.getDefault()));
-            operatorFile = operatorFiles.get(ptOp.getId());
+      poolExecutor.submit(new Runnable()
+      {
+        @Override
+        public void run()
+        {
+          try {
+            containerFile.append(sca.getContainerInfo());
```
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java]: Please check this operation is thread safe. There can be multiple runnables executing concurrently.

Comment (davidyan74) [inline engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java]: the append method of FSJsonLineFile is declared synchronized.

**Pull request merged**